crates/red_knot_python_semantic/src/types.rs

```diff
@@ -120,6 +120,13 @@ fn symbol_by_id<'db>(db: &'db dyn Db, scope: ScopeId<'db>, symbol: ScopedSymbolI
 
 /// Shorthand for `symbol_by_id` that takes a symbol name instead of an ID.
 fn symbol<'db>(db: &'db dyn Db, scope: ScopeId<'db>, name: &str) -> Symbol<'db> {
+    match file_to_module(db, scope.file(db)) {
+        Some(module) if module.name().as_str() == "typing" && name == "TYPE_CHECKING" => {
```

Comment:
Worth adding a comment that `typing_extensions` just re-exports the symbol from `typing`?